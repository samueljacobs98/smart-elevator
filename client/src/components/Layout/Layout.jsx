import "./Layout.scss";

const Layout = ({ isLoading, children }) => {
  return (
    <main className={`layout ${isLoading && "layout--loading"}`}>
      {children}
    </main>
  );
};

export default Layout;
