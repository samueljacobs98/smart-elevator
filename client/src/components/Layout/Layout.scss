.layout {
  height: 100vh;
  display: grid;
  grid-template-columns: 1fr 1fr;
  grid-template-areas:
    "first first"
    "second third";

  &--loading {
    display: flex;
    justify-content: center;
    align-items: center;
  }

  & > :nth-child(1) {
    grid-area: first;
  }

  & > :nth-child(2) {
    grid-area: second;
  }

  & > :nth-child(3) {
    grid-area: third;
  }
}
