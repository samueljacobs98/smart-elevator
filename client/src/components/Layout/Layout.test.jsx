import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import Layout from "./Layout";

describe("Layout", () => {
  const renderComponent = (children) => render(<Layout>{children}</Layout>);

  it("renders without crashing", () => {
    renderComponent();
    expect(screen).toBeDefined();
  });

  it("renders child content correctly", () => {
    const content = "Content";

    renderComponent(content);

    const childElement = screen.getByText(content);
    expect(childElement).toBeInTheDocument();
  });

  it("renders multiple-element child content correctly", () => {
    const content = ["Content 1", "Content 2"];
    const children = content.map((text) => <div key={text}>{text}</div>);

    renderComponent(children);

    content.forEach((text) => {
      const childElement = screen.getByText(text);
      expect(childElement).toBeInTheDocument();
    });
  });

  it("should match snapshot", () => {
    const { container } = renderComponent();

    expect(container).toMatchSnapshot();
  });
});
