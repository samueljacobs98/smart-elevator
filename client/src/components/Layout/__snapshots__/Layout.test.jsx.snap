// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout should match snapshot 1`] = `
<div>
  <main
    class="layout"
  />
</div>
`;
